pkg/api/entries.go

```diff
@@ -301,9 +306,64 @@ func createLogEntry(params entries.CreateLogEntryParams) (models.LogEntry, middl
 	logEntry := models.LogEntry{
 		entryID: logEntryAnon,
 	}
+
+	if api.newEntryPublisher != nil {
+		// Publishing notifications should not block the API response.
+		go func() {
+			var subjects []string
+			verifier, err := entry.Verifier()
+			if err != nil {
+				log.ContextLogger(ctx).Warnf("Could not get verifier for log entry %s: %w", entryID, err)
+			} else {
+				subjects = verifier.Subjects()
+			}
+			pbEntry, err := tle.GenerateTransparencyLogEntry(logEntryAnon)
+			if err != nil {
+				incPublishEvent(newentry.Name, "", false)
+				log.ContextLogger(ctx).Error(err)
+				return
+			}
+			event, err := newentry.New(entryID, pbEntry, subjects)
+			if err != nil {
+				incPublishEvent(newentry.Name, "", false)
+				log.ContextLogger(ctx).Error(err)
+				return
+			}
+			if viper.GetBool("rekor_server.publish_new_entry_protobuf") {
+				go publishEvent(ctx, api.newEntryPublisher, event, events.ContentTypeProtobuf)
```

Comment:
Given this is already in a go routine, do we need another?

Reply:
Having these start in different Go routines is now important to have the RLocks() called as soon as possible to help with safe shutdown.